source/geh_calculated_measurements/tests/electrical_heating/electrical_heating_tests/infrastructure/electricity_market/test_repository.py

```diff
@@ -0,0 +1,65 @@
+import pytest
+
+from geh_calculated_measurements.electrical_heating.infrastructure import (
+    ElectricityMarketRepository,
+)
+
+
+def test__when_consumption_missing_expected_column_raises_exception(
+    electricity_market_repository_missing_col: ElectricityMarketRepository, monkeypatch
+) -> None:
+    # Act/Assert
+
+    with pytest.raises(ValueError, match=r"Column has_electrical_heating not found in CSV"):
+        electricity_market_repository_missing_col.read_consumption_metering_point_periods()
```

Comment:
No need for the AAA pattern in this small test.
Please note that this and others of my comments apply to more than one test.
```suggestion
def test__when_consumption_missing_expected_column_raises_exception(
    electricity_market_repository_missing_col: ElectricityMarketRepository, monkeypatch
) -> None:
    with pytest.raises(ValueError, match=r"Column has_electrical_heating not found in CSV"):
        electricity_market_repository_missing_col.read_consumption_metering_point_periods()
```